### PR TITLE
CI: Update micro versions of Python

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

File: cupy/_core/core.pyx [hunk 1]
```diff
@@ -1208,7 +1208,7 @@ cdef class _ndarray_base:
             if op == 2:
                 # cupy.ndarray does not support dtype=object, but
                 # allow comparison with None, Ellipsis, and etc.
-                if type(other).__eq__ is object.__eq__:
+                if type(other).__eq__ is object.__eq__ or other is None:
```
Comment (asi1024) [inline cupy/_core/core.pyx:1211]: note:
Python 3.12.0
```py
>>> type(None).__eq__
<slot wrapper '__eq__' of 'object' objects>
```

Python 3.12.4
```py
>>> type(None).__eq__
<slot wrapper '__eq__' of 'NoneType' objects>
```

**Comment by kmaehashi** [comment]
Looks good, please regenerate and then run `/test full`.

**Review by takagi — APPROVED** [review]
LGTM!

**Pull request merged**